src/util/address.rs

```diff
@@ -727,6 +727,23 @@ impl Address {
             (Network::Testnet, _) | (Network::Regtest, _) | (Network::Signet, _) => true
         }
     }
+
+    /// Determine if an address is related to a given secp256k1 public key
```

Comment:
The documentation should explain better what "related to public key" means. I suggest enumerating all possibilities and explaining each.